### PR TITLE
Added Team Member API

### LINKED ISSUE #1822: User management from API
Hello,

Since the LDAP auth approach was scrubbed is there any possibility that we as an additional feature
could add the ability to manage users from the API?
Having a large organization and being able to integrate it with other systems is crucial or most enterprises.

Best regards

### PR DESCRIPTION
**Pull request by jbrooksuk**
Closes #1822

### CODE REVIEW

File: app/Bus/Commands/User/AddTeamMemberCommand.php [hunk 1]
```diff
@@ -48,8 +48,9 @@
      */
     public $rules = [
         'name'     => 'required|string',
-        'password' => 'string',
-        'level'    => 'int',
+        'email'    => 'required|email',
+        'password' => 'required|string',
+        'level'    => 'int|min:1|max:2',
```
Comment (GrahamCampbell) [inline app/Bus/Commands/User/AddTeamMemberCommand.php]: In all our rules on L5.3, if they're not required, we need to add `nullable` to them.

Comment (GrahamCampbell) [inline app/Bus/Commands/User/AddTeamMemberCommand.php]: That's something I didn't upgrade, but need to do.

Comment (jbrooksuk) [inline app/Bus/Commands/User/AddTeamMemberCommand.php]: I'll do that as a separate PR later.

**Comment by GrahamCampbell** [comment]
Needs retargetting to "develop".

**Pull request closed without merging**